### PR TITLE
feat: training code

### PR DESCRIPTION
**Pull request by Yhuley**
My training api extended with Post entity

### CODE REVIEW

File: training/src/controllers/teams/handlers/createTeam.handler.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export const createTeam = async (body: TeamBody, userId: string): Promise<Team> => {
+  const em = RequestContext.getEntityManager();
+
+  // create team
+  const team = em.create(Team, body);
+  await em.persistAndFlush(team);
+
+  // create team member
+  const membership = em.create(Membership, { userId, teamId: team.id })
+  await em.persistAndFlush(membership);
```
Comment (kkurylchenko) [inline training/src/controllers/teams/handlers/createTeam.handler.ts:15]: You can use Flush once `await em.persistAndFlush([team, membership]);`

File: training/src/controllers/teams/handlers/getTeams.handler.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Membership } from './../../../entities/membership.entity';
+import { RequestContext } from '@mikro-orm/core';
+import { PostgreSqlDriver, SqlEntityManager } from '@mikro-orm/postgresql';
+
+export const getTeams = async (userId: string) => {
+  const em = RequestContext.getEntityManager() as SqlEntityManager<PostgreSqlDriver>;
+
+  const res = await em.createQueryBuilder(Membership, 'm')
```
Comment (kkurylchenko) [inline training/src/controllers/teams/handlers/getTeams.handler.ts:8]: use mikro orm

File: training/src/controllers/teams/team.controller.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  @Patch('/:id')
+  @Authorized()
+  @OpenAPI({ summary: 'Update team' })
+  @Representer(TeamView)
+  async updateTeam(@Param("id") id: string, @Body({}, { skipMissingProperties: true }) body: TeamBody) {
+    return updateTeam(id, body);
+  }
+
+  @Delete('/:id')
+  @Authorized()
```
Comment (kkurylchenko) [inline training/src/controllers/teams/team.controller.ts:47]: If authorization rules are the same for all endpoints you can move it to the controller level

File: training/src/entities/answer.entity.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import { Offer } from "./offer.entity";
+import { Question } from "./question.entity";
+
+@Entity()
+export class Answer extends BaseEntity<Answer, 'id'> {
+  @PrimaryKey({ columnType: 'uuid' })
+  id: string = v4();
+
+  @Property()
+  public answer: string;
```
Comment (kkurylchenko) [inline training/src/entities/answer.entity.ts:12]: I would rename it, its a bit confusing to have `answer` field inside the answer entity

File: training/src/entities/answer.entity.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  @ManyToOne(() => Offer, { name: 'offer_id' })
+  public offerId: string;
+
+  @ManyToOne(() => Question, { name: 'question_id' })
+  public questionId: string;
```
Comment (kkurylchenko) [inline training/src/entities/answer.entity.ts:18]: Answers should be removed when offers or questions are deleted

File: training/src/controllers/requests/handlers/getRequest.handler.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { RequestContext } from '@mikro-orm/core';
+import { Request } from '../../../entities/request.entity';
+
+export const getRequest = async (id: string) => {
+  const em = RequestContext.getEntityManager();
+
+  const request = await em.findOneOrFail(Request, { id });
+  await em.populate(request, ['tender']);
```
Comment (ljobse) [inline training/src/controllers/requests/handlers/getRequest.handler.ts:8]: In simple cases populate can just be included in the find, no need for the extra statement

**Pull request closed without merging**